### DISCUSSION TITLE
Support for MariaDB: whereJsonContains, etc.

### DISCUSSION THREAD

**Discussion by wq9578** [laravel/framework#52054] [General] [answered]
MariaDB is supposed to be a drop-in replacement for MySQL.

However, some Laravel features are implemented for MySQL, but not for MariaDB, according to the [docs](https://laravel.com/docs/11.x/queries#json-where-clauses):

> JSON Where Clauses
> 
> Laravel also supports querying JSON column types on databases that provide support for JSON column types. Currently, this includes MySQL 8.0+, PostgreSQL 12.0+, SQL Server 2017+, and SQLite 3.39.0+ (with the JSON1 extension).

Why is this the case? Is there a prerequisite missing from MariaDB, or is the feature just not implemented by Laravel yet?

As MariaDB is supposed to ensure a free SQL implementation in the open source space, it would be desirable for Laravel have full MariaDB support.

**Comment by staudenmeir** [comment, marked as answer]
Hi @wq9578,
Good catch. When this part of the docs was written, Laravel didn't officially support MariaDB. That's why it isn't mentioned.

Since then, there have been multiple PRs to support MariaDB. I'll check again, but all these JSON queries should also work with MariaDB.

**Reply by wq9578** [reply]
Thanks!

A follow-up question: Is PostgreSQL supported the same way MySQL is? I'm considering migrating from MySQL to PostgreSQL at some point.
Any other reasons not to choose PostgreSQL for a Laravel project?
